browser/views.py

```diff
@@ -681,27 +700,49 @@ def edit_investigator_record(request,person_id,research_id):
     person = get_object_or_404(Person, pk=person_id)
     research = get_object_or_404(Investigator, pk=research_id)
     template = "browser/investigator.html"
+    institution_id = research.institution.id if research.institution else None
 
     # Retreive investigator data already stored in database and display to the user
     form = InvestigatorForm(initial={'subject': research.subject, 'role': research.role,
-                                             'year': research.year})
+                                             'year': research.year, 'institution_search': research.institution,
+                                     'institution_id': institution_id})
```

Comment:
this is only necessary in a GET request, isn't it? It should only be executed in a GET request to avoid unnecessary database calls and code execution.